### ISSUE TITLE
`:redundant-let` linter not working inside `deftest` when it has a custom hook 

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#2215]
**version**

```
clj-kondo v2023.09.07
```

**platform**

Native

**problem**

We have a custom `:analyze-call` hook defined for `deftest` that looks like this:

```clj
(defn deftest  [{:keys [node], :as arg}]
  ;; you can actually comment this line out and the problem still happens
  (deftest-check-not-horrifically-long node)
  arg)
```

With this config, the `:redundant-let` linter does not work inside `deftest` any more:

<img width="219" alt="image" src="https://github.com/clj-kondo/clj-kondo/assets/1455846/db2de7f7-aca3-45cc-884a-493939c1e6f6">

**repro**

```clj
;; .clj-kondo/config.edn
{:hooks {:analyze-call {clojure.test/deftest hooks.clojure.test/deftest}}}
```

```clj
;; .clj-kondo/hooks/clojure/test.clj
(ns hooks.clojure.test)

(defn deftest [arg]
  arg)
```

```clj
;; test/fail.clj
(ns fail
  (:require [clojure.test :refer [deftest]]))

(deftest my-test
  (let [x 1]
    (let [y 2]
      (assert (not=  x y)))))
```

```sh
# output
clj-kondo --lint test/fail.clj
linting took 7ms, errors: 0, warnings: 0
```

**expected behavior**

Should still warn about redundant-let even if a custom hook is in use

**Comment by borkdude** [comment]
Thanks for reporting, I can successfully reproduce this locally and will have a look.
